### ISSUE TITLE
Implement variability in feature life

### ISSUE THREAD

**Issue by JayToDaBee** [JayToDaBee/TwelveAngryNodes_4.24#47]
We should allow resource life to vary a little bit from the standard, either through a standardized randomization, or json-defined min-max values.

**Comment by JayToDaBee** [comment]
Currently featureLife is also used as a maximum life for a feature. Do we want to allow features to have more than maximum life? Or define maximum life on a per feature instance basis?

**Comment by JayToDaBee** [comment]
After looking at this a bit more, I'm coming to the conclusion that I don't like the idea of having a per-instance max life. This would require to add at least one more field to the FeatureInstance struct per spot, which will be a significant increase in the memory needs of the game, for a relatively minor benefit. I'll look into allowing features to have more than max life.

**Comment by JayToDaBee** [comment]
I have differentiated the 'FullLife' and 'MaxLife' concepts. Full life is the minimum life required for a feature to act to full capacity, and corresponds to the previous 'FeatureLife'. Max life is the maximum amount of life a feature can have, and nothing can make it top that.

Automatic generation of vegetation and deposits now selects a random amount of life that lies between those two. At the moment I did not make it so that an instanced feature has a different life on different spots; the randomization happens per hex (we can always change that later if deemed necessary). Eventually features like vegetation would probably not get a random life, but rather an 'equilibrium' life that depends on tile fertility.
